ks-apiserver failed to start up

<!--
You don't need to remove this comment section, it's invisible on the issues page.

## General remarks

* Attention, please fill out this issues form using English only!
* 注意！GitHub Issue 仅支持英文，中文 Issue 请在 [论坛](https://kubesphere.com.cn/forum/) 提交。
* This form is to report bugs. For general usage questions you can join our Slack channel
       [KubeSphere-users](https://join.slack.com/t/kubesphere/shared_invite/enQtNTE3MDIxNzUxNzQ0LTZkNTdkYWNiYTVkMTM5ZThhODY1MjAyZmVlYWEwZmQ3ODQ1NmM1MGVkNWEzZTRhNzk0MzM5MmY4NDc3ZWVhMjE)
-->

**Describe the Bug**
M1221 10:42:34.026372    1 client_config. go:615] Neither kubeconfig nor master was specified. Using the inclusterConfig. This might not work.
v1221 10:42:34.047376     1 options. go:191) ks-apiserver starts without redis provided, it will use in memory cache. This may cause inconsistencies when running ks-apiserver with multiple replicas.
I1221 10:42:34.047391      1 interface. go:501 start helm repo informer
panic: runtime error: index out of range (o] with lensth 0
goroutine 87 (rumine]:
kubesphere.io/kubesphere/pkg/kapis/servicenesh/metrics /vi alpha2. NewHandler (0x c000586500, 0×4226638, Oxc000397760, Ox41eabad, Oxc00051a1c0, 0xe)
/workspace/pkg/kapis/servicenesh/metrics/vlalpha2/handler.go:51+0×57b
kubesphere.io/kubesphere/pkg/kapis/servicenesh/metries/vlalpha?. AddToContainer(0xc000536500,0xc00100£880,0x4226638,0xc000897760,Ox41eabad,Oxc00051a1c0,Oxc000e96080,0×4169840)
/workspace/pkg/kapis/servicemesh/metrics/vlalpha2/register.go:42+0×fa
kubesphere. io/kubesphere/pkg/apiserver. (*APIServer). installKubeSphereAPIs (0xc000714500, Oxc0009 a2.00)
/workspace/pkg/apiserver/apiserver. go:259 +0×1534
kubesphere.io/kubesphere/pkg/apiserver. (*APIServer). PrepareRun (Oxc000714500, Oxc0009a2c00, Oxc000714500, 0x0)
/workspace/pkg/apiserver/apiserver. go:180 +0x17e
buibesshere. iofksbespherefcnd /ks-zpisezvez /app. zum (Oxc000af9280, Ox41 86278, 0xc000056200, 0x0, 0×0)
/workspace/end/ks-apiserver/app/server. go:141 +0x8d
kubesphere. io/kubesphere/end/ks-apiserver /app. Run. func1 (0xc000af9260, Oxc00011c050, 0xc0009a2640)
/workspace/cd/ks-apiserver /app/server. go: 106 +0×45 created by kubesphere. io/kubesphere/cd/ks-apiserver/app. Run
/workspace/end/Ls-apiserver /app/server. go:105 +0x135
![f8af080995cb94b7d940b31d7a87ae2f](https://user-images.githubusercontent.com/49057502/208815911-41fdf647-09b0-48a0-9a92-c630c5fc1479.png)

**Versions Used**
KubeSphere: v3.3.1
Kubernetes: (If KubeSphere installer used, you can skip this)

--- a/pkg/kapis/servicemesh/metrics/v1alpha2/handler.go
+++ b/pkg/kapis/servicemesh/metrics/v1alpha2/handler.go
@@ -48,15 +48,17 @@ func NewHandler(o *servicemesh.Options, client kubernetes.Interface, cache cache
 	if o != nil && o.KialiQueryHost != "" {
 		sa, err := client.CoreV1().ServiceAccounts(KubesphereNamespace).Get(context.TODO(), KubeSphereServiceAccount, metav1.GetOptions{})
 		if err == nil {
-			secret, err := client.CoreV1().Secrets(KubesphereNamespace).Get(context.TODO(), sa.Secrets[0].Name, metav1.GetOptions{})
-			if err == nil {
-				return &Handler{
-					opt: o,
-					client: kiali.NewDefaultClient(
-						cache,
-						string(secret.Data["token"]),
-						o.KialiQueryHost,
-					),
+			if len(sa.Secrets) > 0 {
+				secret, err := client.CoreV1().Secrets(KubesphereNamespace).Get(context.TODO(), sa.Secrets[0].Name, metav1.GetOptions{})
+				if err == nil {
+					return &Handler{
+						opt: o,
+						client: kiali.NewDefaultClient(
+							cache,
+							string(secret.Data["token"]),
+							o.KialiQueryHost,
+						),
+					}
 				}
 			}
 			klog.Warningf("get ServiceAccount's Secret failed %v", err)

